sound/soc/intel/boards/sof_sdw.c

```diff
@@ -949,16 +949,16 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 			{
 				.direction = {false, true},
 				.dai_name = "cs42l43-dp1",
-				.dai_type = SOF_SDW_DAI_TYPE_JACK,
-				.dailink = {SDW_UNUSED_DAI_ID, SDW_JACK_IN_DAI_ID},
+				.dai_type = SOF_SDW_DAI_TYPE_MIC,
+				.dailink = {SDW_UNUSED_DAI_ID, SDW_DMIC_DAI_ID},
+				.init = sof_sdw_cs42l43_dmic_init,
```

Comment:
@charleskeepax Can we keep the dai type order, and just change the dai_name?
i.e. Something like
```patch
-        .dai_name = "cs42l43-dp1",
+        .dai_name = "cs42l43-dp2",
-        .dai_name = "cs42l43-dp2",
+        .dai_name = "cs42l43-dp1",
```

Reply:
We could, the reason I did it this way round is only the first two DAIs work on topology1 so if I change the DAI order then I can no longer access the 4 channel DAI, which means I can't use all 4 mics. Is there something from your side that makes the new order challenging, or are you just looking to simplify the patch?

Reply:
@charleskeepax The order will affect the dai link creation order. For example, below is the dai link order of the original order.
```
[    4.655113] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW0-Playback-SimpleJack, id 0
[    4.655120] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW0-Capture-SimpleJack, id 1
[    4.655124] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW0-Capture-SmartMic, id 4
[    4.655134] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW1-Playback-SmartAmp, id 2
[    4.655138] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW1-Capture-SmartAmp, id 3
```
And below is the order of the new order
```
[    5.055706] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW0-Playback-SimpleJack, id 0
[    5.055711] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW0-Capture-SmartMic, id 4
[    5.055715] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW0-Capture-SimpleJack, id 1
[    5.055724] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW1-Playback-SmartAmp, id 2
[    5.055728] snd_soc_sof_sdw:init_dai_link: sof_sdw sof_sdw: create dai link SDW1-Capture-SmartAmp, id 3
```
I think the topology1 issue might because the topology is looking for "SDW0-Capture" for both headset mic and DMIC? The same issue will happen on topology2, too.
The reason I prefer not changeing the order are 1. the dai link for headset playback and capture will be created together. 2. It will be easier to see what was changed in the commit.
But both are minor.

Reply:
The topology we are using has the post-fixes in it, so it is looking for SDW0-Capture-SmartMic etc.
Yeah the changing of the order was kinda intentional, the important bit seems to be what "pin" the DAI is connected to on the Intel side which is determined by the DAI order rather than the DAI ID. With only pins 2/3 working. I am happy to either merge this as is, or to wait a couple days whilst I work through getting topology2 going, assuming it doesn't appear to have the same limitations as topology1 I would be happy to repush with the order changed.

Reply:
> The topology we are using has the post-fixes in it, so it is looking for SDW0-Capture-SmartMic etc. Yeah the changing of the order was kinda intentional, the important bit seems to be what "pin" the DAI is connected to on the Intel side which is determined by the DAI order rather than the DAI ID. With only pins 2/3 working. I am happy to either merge this as is, or to wait a couple days whilst I work through getting topology2 going, assuming it doesn't appear to have the same limitations as topology1 I would be happy to repush with the order changed.

I got your issue now. We never tried pin 4 or above in IPC3. @plbossart @ranj063 Do you know if IPC3 supports SDW Pin4?

Reply:
@bardliao @charleskeepax I am not aware of any hardware limitation with Pin4, and I don't think there's any IPC limitation either. We use 8 bits to encode the pin.